blog/tests/views_test.py

```diff
@@ -0,0 +1,115 @@
+"""Tests for views' Blog app."""
+
+from datetime import datetime
+
+from django.contrib.auth.models import User
+from django.test import TestCase
+from django.urls import reverse
+
+from ..models import Post
+
+
+class PostListTest(TestCase):
+    """Tests for Post model."""
+
+    def setUp(self):
+        """Set up non-modifiable objects used by all test methods."""
+        bob = User.objects.create(username='bob')
+        self.post_first = Post.objects.create(
+            author=bob,
+            title='Title Bob',
+            text='Text bob',
+            published_date=datetime(day=8, month=11, year=2018),
+        )
+
+        john = User.objects.create(username='john')
+
+        self.post_second = Post.objects.create(
+            author=john, title='Title John', text='Text John'
+        )
+        self.post_third = Post.objects.create(
+            author=john, title='Title John 2', text='Text John 2'
+        )
+        self.post_third.publish()
+
+    def tearDown(self):
+        """Clean-up test data."""
+        del self.post_first
+        del self.post_second
+        del self.post_third
+
+    def test_post_list(self):
+        """Test show post list on the page."""
+        all_post = list(str(i) for i in Post.objects.all())
+        self.assertTrue(
+            all(
```

Comment:
This check is fairly unstable as it will pass even if all three posts have `'Title Bob'` title. You need to check for specific title order.

Reply:
Fixed